### PR TITLE
Make files available in Docker runner image

### LINKED ISSUE #2278: Files specified under the files target property should be included in the docker image
If you specify something with `files: "my_file.txt"`, and also specifies `docker: true` then those files should be included in the resulting docker image. This requires changing the code-generated Dockerfile to copy over those files.

### PR DESCRIPTION
**Pull request by petervdonovan**
Closes #2278.

## Summary by CodeRabbit

- **New Features**
  - Added functionality to copy user-specified files in Docker generation, enhancing the customization of Dockerfiles.
  
- **Tests**
  - Introduced a new test to verify file content comparison and validation within a containerized runtime environment.

### CODE REVIEW

File: core/src/main/java/org/lflang/generator/docker/DockerGenerator.java [hunk 1]
```diff
@@ -150,6 +152,27 @@ protected String generateCopyOfScript() {
...
   }
 
+  /**
+   * Return zero or more COPY commands to copy files specified using the {@code files} target
+   * property from the builder to the runner.
+   */
+  protected String generateCopyOfUserFiles() {
+    var files = context.getTargetConfig().get(FilesProperty.INSTANCE);
+    if (files == null) {
+      return "";
```
Comment (lhstrh) [inline core/src/main/java/org/lflang/generator/docker/DockerGenerator.java:162]: Maybe something a bit verbose, like this, would be better:
```suggestion
      return  "# (No user-specified files to be copied.)";
```

**Review by lhstrh — COMMENTED** [review]
Other than stated comments/question, this looks like the right solution to me. Thanks for the fix, @petervdonovan!

File: core/src/main/java/org/lflang/generator/docker/DockerGenerator.java [hunk 2]
```diff
@@ -150,6 +152,27 @@ protected String generateCopyOfScript() {
     return "# (No pre-run script provided.)";
   }
 
+  /**
+   * Return zero or more COPY commands to copy files specified using the {@code files} target
+   * property from the builder to the runner.
+   */
+  protected String generateCopyOfUserFiles() {
+    var files = context.getTargetConfig().get(FilesProperty.INSTANCE);
```
Comment (lhstrh) [inline core/src/main/java/org/lflang/generator/docker/DockerGenerator.java:160]: As evidenced by test failure, this won't work for all targets. Put it behind a check `context.getTargetConfig.isSupported(FilesProperty.INSTANCE)`

File: core/src/main/java/org/lflang/generator/docker/DockerGenerator.java [hunk 3]
```diff
@@ -150,6 +152,27 @@ protected String generateCopyOfScript() {
...
+    for (var file : files) {
+      var p = Path.of(file);
+      var name = p.getName(p.getNameCount() - 1);
+      ret.append(
+          String.format(
+              "COPY --from=builder \"lingua-franca/%s/src-gen/%s\" \"./%s\"",
+              context.getFileConfig().name, name, name));
+    }
+    return ret.toString();
+  }
```
[the remark this answers is not in the document]
Comment (petervdonovan) [reply core/src/main/java/org/lflang/generator/docker/DockerGenerator.java]: This comment was already shared by Marten

[the remark this answers is not in the document]
Comment (lhstrh) [reply core/src/main/java/org/lflang/generator/docker/DockerGenerator.java]: :rofl: Code Rabbit gave the same feedback as I did...

**Review by lhstrh — APPROVED** [review]
:+1:

**Pull request merged**